pkg/volume/util/fsquota/quota_linux.go

```diff
@@ -439,3 +438,41 @@ func ClearQuota(m mount.Interface, path string) error {
 	}
 	return nil
 }
+
+// RecordProjectIDOnProjectFiles -- Records project file for the given directory-path to /etc/projects and /etc/projectid
+func RecordProjectIDOnProjectFiles(podVolumeDir string) error {
+	id, err := getApplier(podVolumeDir).GetQuotaOnDir(podVolumeDir)
+	if err == nil {
+		fProjects, fProjid, err := openAndLockProjectFiles()
+		if err != nil {
+			defer closeProjectFiles(fProjects, fProjid)
+			list := readProjectFiles(fProjects, fProjid)
+			projectList := list.projects
+			projectIDList := list.projid
+			foundInProjectList, foundInProjectIDList := false, false
+
+			for _, project := range projectList {
+				if project.id == id && project.data == podVolumeDir {
```

Comment:
* The case in which project.data matches but not project.id, I am wondering if it's possible, because `GetQuotaOnDir` returns the current id applied to the directory.
* If the project.id matches but the project.data doesn't mean the current project.id is assigned to multiple directories, so we should add a new entry in the projectList for the current `id` and `podVolumeDir` which is happening in the current implementation.

Am I missing some case here?